redfish-core/lib/pcie.hpp

```diff
@@ -423,188 +423,193 @@ inline void requestRoutesSystemPCIeDeviceCollection(App& app)
-                            {"Id", device}};
-
-                        if (std::string* property = std::get_if<std::string>(
-                                &pcieDevProperties["Manufacturer"]);
-                            property)
-                        {
-                            asyncResp->res.jsonValue["Manufacturer"] =
-                                *property;
-                        }
-
-                        if (std::string* property = std::get_if<std::string>(
-                                &pcieDevProperties["DeviceType"]);
-                            property)
-                        {
-                            if (!property->empty())
-                            {
-                                asyncResp->res.jsonValue["DeviceType"] =
-                                    *property;
-                            }
-                        }
-
-                        if (std::string* property = std::get_if<std::string>(
-                                &pcieDevProperties["GenerationInUse"]);
-                            property)
-                        {
-                            std::string generation =
-                                analysisGeneration(*property);
-                            if (!generation.empty())
-                            {
-                                asyncResp->res
-                                    .jsonValue["PCIeInterface"]["PCIeType"] =
-                                    generation;
-                            }
-                        }
+                BMCWEB_LOG_DEBUG
+                    << "failed to get PCIe Device properties ec: " << ec.value()
+                    << ": " << ec.message();
+                if (ec.value() ==
+                    boost::system::linux_error::bad_request_descriptor)
+                {
+                    messages::resourceNotFound(asyncResp->res, "PCIeDevice",
+                                               device);
+                }
+                else
+                {
+                    messages::internalError(asyncResp->res);
+                }
+                return;
+            }
 
-                        if (std::string* property = std::get_if<std::string>(
-                                &pcieDevProperties["PartNumber"]);
-                            property)
-                        {
-                            if (!property->empty())
-                            {
-                                asyncResp->res.jsonValue["PartNumber"] =
-                                    *property;
-                            }
-                        }
+            asyncResp->res.jsonValue = {
+                {"@odata.type", "#PCIeDevice.v1_9_0.PCIeDevice"},
+                {"@odata.id",
+                 "/redfish/v1/Systems/system/PCIeDevices/" + device},
+                {"Name", "PCIe Device"},
+                {"Id", device}};
+            std::vector<std::string> splitText;
```

Comment:
@gtmills @aahmed-2  main code is 468 to 488 Line

this diff shows a long change because I need `std::string& path = std::get<1>(it->second);` this path on line 437 line. 
And that is why I created a new method because changing line gives more indent. that makes it hard to read